lib/Controller/ChangePasswordController.php

```diff
@@ -185,6 +190,17 @@ public function changePassword($username, $password = null, $recoveryPassword =
 
 		// if encryption is disabled, proceed
 		try {
+			//Check if the user is created using email, and if the user had not logged in once
+			$disablePasswordSet = $this->config->getUserValue($targetUserObject->getUID(), 'owncloud', 'disable_password_reset', 'false');
+			if (($targetUserObject->getLastLogin() === 0) && ($disablePasswordSet === 'true')) {
```

Comment:
is this a requirement ?

I think the admin should still be able to set a different password even after the email is out. Setting a new password would invalidate the token (to be checked). This could happen in a scenario where the email was sent to a wrong or compromised email address.

Reply:
Removed it. I will create a PR in the core, where the password setting triggered by user will be blocked if the user is not logged in once.